src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -32,6 +33,23 @@
     /**
      * {@inheritdoc}
      */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $intersect = array_intersect_assoc(
+            $this->configuration['annotation-white-list'],
+            $this->configuration['annotation-black-list']
+        );
+
+        if (count($intersect)) {
+            throw new InvalidFixerConfigurationException($this->getName(), sprintf('Annotation cannot be used in both the white- and black list, got duplicates: "%s".', implode('", "', array_keys($intersect))));
```

Comment:
shouldn't this be on config-definition level ?

Reply:
when validating one configuration option you don't have access to the other options to check for conflicts, I think.

Reply:
you could have
first param of normalizer option is
`The closure is invoked when {@link resolve()} is called. The closure has access to the resolved values of other options through the passed {@link Options} instance.`

Reply:
grep for `->setNormalizer` in our project

Reply:
I don't see how it would work.
This tests if the two normalized option values do not conflict. Within both the `setAllowedValues` and `setNormalizer` callbacks there is no access to the both the normalized values.
This could _only_ work if there was some way to add a hook that is called after the normalization of all options has been done, but I don't know of such hook.

Reply:
hook for `setNormalizer` got 2 params, first is all options, so we have access to option we want to cross-validate